external/js/cothority/src/byzcoin/client-transaction.ts

```diff
@@ -364,6 +366,73 @@ export class Instruction extends Message<Instruction> {
         });
         return h.digest();
     }
+
+    /**
+     * Return a general-purpose json representation of an instruction that is
+     * contracts aware, ie. knows how to render specific contracts arguments.
+     * 
+     * The JSON has the following structure:
+     * 
+     *  {
+     *      type: "spawn|invoke|delete",
+     *      contract: "config|darc|..."
+     *      args: [
+     *          {
+     *              "name": "<NAME>",
+     *              "value": "<VALUE>",    // Arg value or its summary
+     *              "full": "<FULL VALUE>" // Optional. useful if the argument
+     *                                     // have a very long representation.
+     *          }
+     *          ...
+     *      ]
+     *  }
+     * 
+     */
+    getArranged(): { [k: string]: any } {
+        let res: {[k: string]: any} = {};
+
+        switch (this.type) {
+            case Instruction.typeSpawn:
+                res.type = "spawn";
+                res.args = [];
+
+                switch (this.spawn.contractID) {
+                    case "config":
+                        res.contract = "config";
+                        res.args = Config.Spawn(this.spawn.args)
+                        break;
+                    case "darc":
+                        res.contract = "darc";
+                        res.args = Darc.Spawn(this.spawn.args)
+                        break;
+                }
+
+                break;
+
+            case Instruction.typeInvoke:
+                res.type = "invoke";
+                res.args = [];
+
+                switch (this.invoke.contractID) {
+                    case "config":
+                        res.contract = "config";
+                        res.args = Config.Invoke(this.invoke.args)
+                        break;
+                    case "darc":
+                        res.contract = "darc";
+                        res.args = Darc.Invoke(this.invoke.args)
+                        break;
+                }
+
+                break;
+
+            case Instruction.typeDelete:
+                res.type = "delete";
+                break;
+        }
```

Comment:
Missing `default` case: there should be either a "unknown class", or even an error thrown. Else somebody using the method might be surprised that it doesn't work...

Reply:
Now it's cleaned up, so it's OK. But in general we should try to break code culture only when it goes into 'best practices'. And using `status` instead of `throw new Error` is not a best practice.